cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestBase.java

```diff
@@ -53,29 +59,110 @@
 import org.junit.Before;
 import org.junit.ClassRule;
 import org.junit.rules.TemporaryFolder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.io.File;
 import java.io.IOException;
 import java.net.URI;
+import java.net.URL;
 import java.util.List;
 import java.util.Map;
+import java.util.Properties;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
 import javax.annotation.Nullable;
+import javax.ws.rs.ServiceUnavailableException;
 
 /**
  *
  */
 public class IntegrationTestBase {
+  private static final Logger LOG = LoggerFactory.getLogger(IntegrationTestBase.class);
 
   @ClassRule
   public static final TemporaryFolder TEMP_FOLDER = new TemporaryFolder();
 
   @Before
   public void setUp() throws Exception {
+    checkSystemServices();
+    assertUnrecoverableResetEnabled();
     assertIsClear();
   }
 
+  private void checkSystemServices() throws ServiceUnavailableException {
+    MonitorClient monitorClient = getMonitorClient();
+
+    try {
+      checkSystemServicesOk(monitorClient);
+    } catch (UnauthorizedException e) {
+      LOG.info("Caught UnAuthorizedException, Getting Credentials");
+      // security is enabled, we need to get access token before checking system services
+      URL url = null;
+      try {
+        Properties properties = new Properties();
+        properties.setProperty("security.auth.client.username", System.getProperty("userName"));
+        properties.setProperty("security.auth.client.password", System.getProperty("password"));
```

Comment:
On second thoughts - let's keep the system properties. We can always set environment variable, and use that to set the system property. System properties are more flexible.